src/emulator/functionsEmulatorUtils.ts

```diff
@@ -63,3 +72,47 @@ export function removePathSegments(path: string, count: number): string {
     .slice(count)
     .join("/");
 }
+
+/**
+ * Parse a semver version string into parts, filling in 0s where empty.
+ */
+export function parseVersionString(version?: string): ModuleVersion {
+  const parts = (version || "0").split(".");
+
+  // Make sure "parts" always has 3 elements. Extras are ignored.
+  parts.push("0");
+  parts.push("0");
+
+  return {
+    major: parseInt(parts[0], 10),
+    minor: parseInt(parts[1], 10),
+    patch: parseInt(parts[2], 10),
+  };
+}
+
+/**
+ * Compare two SemVer version strings.
+ *
+ * Returns:
+ *   - Positive number if a is greater.
+ *   - Negative number if b is greater.
+ *   - Zero if they are the same.
+ */
+export function compareVersionStrings(a?: string, b?: string) {
+  const versionA = parseVersionString(a);
+  const versionB = parseVersionString(b);
+
+  if (versionA.major != versionB.major) {
+    return versionA.major - versionB.major;
+  }
+
+  if (versionA.minor != versionB.minor) {
+    return versionA.minor - versionB.minor;
+  }
+
+  if (versionA.patch != versionB.patch) {
+    return versionA.patch - versionB.patch;
+  }
+
+  return 0;
+}
```

Comment:
I feel like you might be reinventing the wheel here a bit. [semver](https://www.npmjs.com/package/semver) should do this for you, and it's already in the repo!

Reply:
(ohh, but I see the note at the top of the file now... hmmm.... your choice.)

Reply:
We can't import `semver` into the runtime so the wheel reinvention is on purpose.